### ISSUE TITLE
Conditional formatting overwrites the format when no format is set

### ISSUE THREAD

**Issue by ndench** [PHPOffice/PhpSpreadsheet#3202]
This is:

```
- [x] a bug report
- [ ] a feature request
- [ ] **not** a usage question (ask them on https://stackoverflow.com/questions/tagged/phpspreadsheet or https://gitter.im/PHPOffice/PhpSpreadsheet)
```

### What is the expected behavior?

Conditional formats should not be changed when reading then writing an excel file.

### What is the current behavior?

Reading an excel file that contains a conditional formatting rule, where the format is `No format set`, and then writing that file immediately results in the rule being changed to have a format.

Having `No format set` is a valid rule because it's the easiest way to not show any conditional formatting when the cell is blank.
ie. Have a rule for `value > 0`, another for `value < 0`, then have a `No format set` for `value is blank` that has `stop if true` enabled.
See "Method 1" on [this post](https://www.extendoffice.com/documents/excel/2499-excel-conditional-formatting-ignore-blank-zero.html).

### What are the steps to reproduce?

```php
<?php

require __DIR__ . '/vendor/autoload.php';

// Read the excel file that has a single formatted cell
$reader = new \PhpOffice\PhpSpreadsheet\Reader\Xlsx();
$excel = $reader->load('no-format.xlsx');

// Immediately write the excel file back out without making any changes
$writer   = new \PhpOffice\PhpSpreadsheet\Writer\Xlsx($excel);
$writer->save('no-format-output.xlsx');
```

Here is the original file that has no content, and only a a single conditional formatting rule for `A1 contains a blank value` with the format `Not format set`
[no-format.xlsx](https://github.com/PHPOffice/PhpSpreadsheet/files/10073076/no-format.xlsx)
![no-format](https://user-images.githubusercontent.com/2062388/203481552-f1e40658-f1b3-4fd6-8c65-f86debdb8dfb.png)

Here is the output file after running the above script, you can see the conditional formatting rule now has a format of `light red fill, dark red text`.
[no-format-output.xlsx](https://github.com/PHPOffice/PhpSpreadsheet/files/10073104/no-format-output.xlsx)

![no-format-output](https://user-images.githubusercontent.com/2062388/203481733-ef2d66c7-f71c-4e28-967d-21b78b920277.png)

### What features do you think are causing the issue

- [ ] Reader
- [x] Writer
- [ ] Styles
- [ ] Data Validations
- [ ] Formula Calculations
- [ ] Charts
- [ ] AutoFilter
- [ ] Form Elements

### Does an issue affect all spreadsheet file formats? If not, which formats are affected?

- XLSX

### Which versions of PhpSpreadsheet and PHP are affected?

- PHP 8.1.2
- PhpSpreadsheet 1.25.2 (1.23.0 also affected)

### Anything else?

I love PhpSpreadsheet and I'm more than happy to help debug this further/fix the issue if someone can point me in the right direction. I've tried stepping through with xdebug but it's kinda hard to follow and locate the issue.

**Comment by MarkBaker** [comment]
This is new for me, I hadn't realised that this was possible in Excel.
When we create a Conditional Formatting Rule, we always create a Style object, whether when loading a file, or when creating a CF Rule within PHP Code; and the Writers expect that Style object when saving the file.

Clearly this is a bug; and the Style should be nullable. It's a fix that will have to be done in several small steps; the internal representation will have to allow for nullable style, and all of the Writers as step #1, with the Readers as step #2, and these can be done individually - so it will likely be at least 2 releases before we can implement this fully.

**Comment by ndench** [comment]
Thanks for the response @MarkBaker. Are these changes something I can help with?
